### PR TITLE
Fix #757 - added empty field validation in MerchantTransferActivity

### LINKED ISSUE #757: No empty field validation in MerchantTransferActivity
**Summary:**
MerchantTransferActivity does not validate for empty amount field

### PR DESCRIPTION
**Pull request by dubeyaayush07**
Fixes #757

- [x] Apply the `AndroidStyle.xml` style template to your code in Android Studio.

- [x] Run the unit tests with `./gradlew check` to make sure you didn't break anything

- [x] If you have multiple commits please combine them into one commit by squashing them.

### CODE REVIEW

File: mifospay/src/main/java/org/mifos/mobilewallet/mifospay/merchants/ui/MerchantTransferActivity.java [hunk 1]
```diff
@@ -137,14 +137,16 @@ public void onSlide(@NonNull View view, float v) {
     @OnClick(R.id.btn_submit)
     public void makeTransaction() {
         String externalId = tvMerchantVPA.getText().toString().trim();
-        String eamount = etAmount.getText().toString().trim();
+        String amount = etAmount.getText().toString().trim();
 
-        double amount = Double.parseDouble(eamount);
-        if (amount <= 0) {
+        if (amount.equals("")) {
+            showToast(Constants.PLEASE_ENTER_ALL_THE_FIELDS);
```
Comment (jawidMuhammadi) [inline mifospay/src/main/java/org/mifos/mobilewallet/mifospay/merchants/ui/MerchantTransferActivity.java]: For translation purposes, we have to add any strings in the string.xml folder. If you put the string inside the java classes then it is not possible to translate in other languages.

**Review by jawidMuhammadi — CHANGES_REQUESTED** [review]

File: mifospay/src/main/java/org/mifos/mobilewallet/mifospay/merchants/ui/MerchantTransferActivity.java [hunk 2]
```diff
@@ -137,14 +137,16 @@ public void onSlide(@NonNull View view, float v) {
...
+            return;
+        } else if (Double.parseDouble(amount) <= 0) {
             showToast(Constants.PLEASE_ENTER_VALID_AMOUNT);
             return;
```
Comment (jawidMuhammadi) [inline mifospay/src/main/java/org/mifos/mobilewallet/mifospay/merchants/ui/MerchantTransferActivity.java]: Same here

**Review by jawidMuhammadi — CHANGES_REQUESTED** [review]

File: mifospay/src/main/java/org/mifos/mobilewallet/mifospay/merchants/ui/MerchantTransferActivity.java (hunk 1 above)
Comment (dubeyaayush07) [reply mifospay/src/main/java/org/mifos/mobilewallet/mifospay/merchants/ui/MerchantTransferActivity.java]: I only used the strings which were already defined. Maybe propose this change in another PR

Comment (jawidMuhammadi) [reply mifospay/src/main/java/org/mifos/mobilewallet/mifospay/merchants/ui/MerchantTransferActivity.java]: Yeah, that will be great.

File: mifospay/src/main/java/org/mifos/mobilewallet/mifospay/merchants/ui/MerchantTransferActivity.java [hunk 3]
```diff
@@ -137,14 +137,16 @@ public void onSlide(@NonNull View view, float v) {
     @OnClick(R.id.btn_submit)
     public void makeTransaction() {
         String externalId = tvMerchantVPA.getText().toString().trim();
-        String eamount = etAmount.getText().toString().trim();
+        String amount = etAmount.getText().toString().trim();
 
-        double amount = Double.parseDouble(eamount);
-        if (amount <= 0) {
+        if (amount.equals("")) {
```
Comment (luckyman20) [inline mifospay/src/main/java/org/mifos/mobilewallet/mifospay/merchants/ui/MerchantTransferActivity.java]:
```suggestion
        if (amount.isEmpty()) {
```

**Review by luckyman20 — CHANGES_REQUESTED** [review]

**Comment by dubeyaayush07** [comment]
@luckyman20 done

**Review by devansh-299 — APPROVED** [review]
LGTM

**Review by shiv07tiwari — APPROVED** [review]

**Pull request merged**